crates/astria-sequencer-relayer/src/relayer/write.rs

```diff
@@ -378,6 +472,62 @@ impl<'a> std::fmt::Display for ReportSequencerHeights<'a> {
     }
 }
 
+/// Currently running conversions of Sequencer blocks to Celestia blobs.
+///
+/// The conversion result will be returned in the order they are pushed
+/// into this queue.
+///
+/// Note on the implementation: the conversions are done in a block tokio
+/// task so that conversions are started immediately without needing extra
+/// polling. This means that the only contribution that `FuturesOrdered`
+/// provides is ordering the conversion result by the order the blocks are
+/// received. This however is desirable because we want to submit sequencer
+/// blocks in the order of their heights to Celestia.
+struct Conversions {
+    // The currently active conversions.
+    active: FuturesOrdered<BoxFuture<'static, (SequencerHeight, eyre::Result<Vec<Blob>>)>>,
+
+    // The maximum number of conversions that can be active at the same time.
+    max_conversions: usize,
+}
+
+impl Conversions {
+    fn new(max_conversions: usize) -> Self {
+        Self {
+            active: FuturesOrdered::new(),
+            max_conversions,
+        }
+    }
+
+    fn has_capacity(&self) -> bool {
+        self.active.len() < self.max_conversions
+    }
+
+    fn push(&mut self, block: SequencerBlock) {
+        let height = block.height();
+        let fut = async move {
+            let res = tokio::task::spawn_blocking(move || convert(block)).await;
```

Comment:
does this still start the task immediately even if it's inside an `async move` scope?

Reply:
That's a pretty good question actually!

I think it's indeed only started after the future is polled for the first time (so once the select macro is entered again).

I'm going to move the task spawn outside the block to start the task immediately. This could lead to funny gotchas otherwise.

Thank you!